src/react/accordion/Accordion.tsx

```diff
@@ -0,0 +1,75 @@
+'use client'
+
+import clsx from 'clsx'
+import type { JSX, ReactElement } from 'react'
+import { Children, forwardRef } from 'react'
+
+import { mergeRefs } from 'react-merge-refs'
+import { useMODUKComponent } from '../hooks/useMODUKComponent'
+import { AccordionContext } from './AccordionContext'
+import type { AccordionHeadingTag } from './AccordionHeadingTag'
+import type { AccordionItemProps } from './AccordionItem'
+import { AccordionItemIndexContext } from './AccordionItemIndexContext'
+
+interface AccordionProps {
+  children: ReactElement<AccordionItemProps>[] | ReactElement<AccordionItemProps>
+  headingTag?: AccordionHeadingTag
+  hideAllSectionsText?: string
+  hideSectionText?: string
+  hideSectionAriaLabelText?: string
+  id: string
+  rememberExpanded?: boolean
+  showAllSectionsText?: string
+  showSectionAriaLabelText?: string
+  showSectionText?: string
+}
+
+export const Accordion = forwardRef<HTMLDivElement, AccordionProps & JSX.IntrinsicElements['div']>(({
+  children,
+  className,
+  headingTag = 'h2',
+  hideAllSectionsText,
+  hideSectionText,
+  hideSectionAriaLabelText,
+  id,
+  rememberExpanded,
+  showAllSectionsText,
+  showSectionText,
+  showSectionAriaLabelText,
+  ...rest
+}, forwardedRef) => {
+  const { ref } = useMODUKComponent('Accordion')
+
+  const i18nAttributes = {
+    'data-i18n.hide-all-sections': hideAllSectionsText,
+    'data-i18n.hide-section': hideSectionText,
+    'data-i18n.hide-section-aria-label': hideSectionAriaLabelText,
+    'data-i18n.show-all-sections': showAllSectionsText,
+    'data-i18n.show-section': showSectionText,
+    'data-i18n.show-section-aria-label': showSectionAriaLabelText,
+  }
+
+  return (
+    <AccordionContext.Provider value={{ headingTag, id }}>
+      <div
+        ref={mergeRefs([ref, forwardedRef])}
+        key={Children.count(children)}
```

Comment:
As per the original spike – force the component to remount (so the JS reruns)  if the number of items changes.